src/de/mod.rs

```diff
@@ -1,4 +1,209 @@
 //! Serde `Deserializer` module
+//!
+//! # Difference between `#text` and `#content` special names
+//!
+//! quick-xml supports two special names for fields -- `#text` and `#content`.
+//! Although they may seem the same, there is a distinction. Two different
+//! names is required mostly for serialization, because quick-xml should know
+//! how you want to serialize certain constructs, which could be represented
+//! through XML in multiple different ways.
+//!
+//! The only difference in how complex types and sequences are serialized.
+//!
+//! ## `#text`
+//! `#text` is used when you want to write your XML as a text or a CDATA content.
+//! More formally, field with that name represents simple type definition with
+//! `{variety} = atomic` or `{variety} = union` whose basic members are all atomic,
+//! as described in the [specification].
+//!
+//! As a result, not all types of such fields can be serialized. Only serialization
+//! of following types are supported:
+//! - all primitive types (strings, numbers, booleans)
+//! - unit variants of enumerations (serializes to a name of a variant)
+//! - newtypes (delegates serialization to inner type)
+//! - [`Option`] of above (`None` serializes to nothing)
+//! - sequences (including tuples and tuple variants of enumerations) of above,
+//!   excluding `None` and empty string elements (because it will not be possible
+//!   to deserialize them back). The elements are separated by space(s)
+//! - unit type `()` and unit structs (serializes to nothing)
+//!
+//! Complex types, such as structs and maps, are not supported in this field.
+//! If you want them, you should use `#content`.
+//!
+//! Sequences serialized to a space-delimited string, that is why only certain
+//! types are allowed in this mode:
+//!
+//! ```
+//! # use serde::{Deserialize, Serialize};
+//! # use quick_xml::de::from_str;
+//! # use quick_xml::se::to_string;
+//! #[derive(Deserialize, Serialize, PartialEq, Debug)]
+//! struct AnyName {
+//!     #[serde(rename = "#text")]
+//!     field: Vec<usize>,
+//! }
+//!
+//! let obj = AnyName { field: vec![1, 2, 3] };
+//! let xml = to_string(&obj).unwrap();
+//! assert_eq!(xml, "<AnyName>1 2 3</AnyName>");
+//!
+//! let object: AnyName = from_str(&xml).unwrap();
+//! assert_eq!(object, obj);
+//! ```
+//!
+//! ## `#content`
+//! Representation of primitive types in `#content` does not differ from they representation
+//! in `#text` field. The difference is how sequences are serialized. `#content` serializes
+//! each sequence item as a separate XML element. The name of that element is taken
+//! from serialized type, and because only `enum`s provide such name (their variant
+//! name), only them should be used in such field.
```

Comment:
```suggestion
//! name), only they should be used for such fields.
```

alternately: only enums should be used for such fields.